File: 백준/Gold/1253. 좋다/좋다.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;

		int N = Integer.parseInt(br.readLine());
		int[] arr = new int[N];
		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < N; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}

		Arrays.sort(arr);
		int res = 0;
		for (int i = 0; i < N; i++) {
			int s = 0, e = N - 1;
			while (s < e) {
				if (s == i) {
					s++;
					continue;
				} else if (e == i) {
					e--;
					continue;
				}

				if (arr[s] + arr[e] < arr[i]) {
					s++;
				} else if (arr[s] + arr[e] > arr[i]) {
					e--;
				} else {
					res++;
					break;
				}
			}
		}

		System.out.println(res);
	}
}

File: 백준/Gold/1253. 좋다/좋다.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))

    val N = br.readLine().toInt()
    val st = StringTokenizer(br.readLine())
    val arr = MutableList(N){st.nextToken().toInt()}

    arr.sort()
    var res = 0
    for (i in 0 until N) {
        var s = 0
        var e = N - 1
        val num = arr[i]

        while (s < e) {
            if (s == i) {
                s++
                continue
            } else if (e == i) {
                e--
                continue
            }

            val sum = arr[s] + arr[e]
            if (sum < num) s++
            else if (sum > num) e--
            else {
                res++
                break
            }
        }
    }

    println(res)
}